ostelco-ios-client/Protocols/ApplePayDelegate.swift

```diff
@@ -139,6 +142,21 @@ extension ApplePayDelegate where Self: PKPaymentAuthorizationViewControllerDeleg
 
     // MARK: - Other Helpers.
 
+    // Create appropriate ApplePayError based on the Error
+    func createApplePayError(_ error: Error) -> ApplePayError {
+        // All generic payment errors are treated as primeAPIError
+        var paymentError = ApplePayError.primeAPIError(error)
+        if let apiError = error as? APIHelper.Error {
+            if case let .jsonError(jsonRequestError) = apiError {
+                // Payment is declined when status = 403 and code = FAILED_TO_PURCHASE_PRODUCT
```

Comment:
Is there a better way to check this ?

Reply:
I would probably `switch` on the error you pass in and return from that switch statement if it's a special case, otherwise return the default value: 

```swift
switch error {
case APIHelper.Error.jsonError(let jsonError): 
    // Handle status logic here, return payment declined if needed otherwise keep going
default: 
   break
}

return ApplePayError.primeAPIError(error)
```